dialog/file.go

```diff
@@ -249,23 +301,55 @@ func showFileDialog(save bool, callback interface{}, parent fyne.Window) {
 	d.win.Resize(size)
 
 	d.win.Show()
+	return d
 }
 
-// ShowFileOpen shows a file dialog allowing the user to choose a file to open.
-// The dialog will appear over the window specified.
-func ShowFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window) {
-	if fileOpenOSOverride(callback, parent) {
+// SetFilter sets a filter for limiting files that can be chosen in the file dialog
+func (fd *FileDialog) SetFilter(filter FileFilter) {
+	fd.filter = filter
+	if fd.dialog != nil {
+		fd.dialog.refreshDir(fd.dialog.dir)
+	}
+}
+
+// Show shows the file dialog
+func (fd *FileDialog) Show() {
+	if !fd.save {
+		if fileOpenOSOverride(fd.callback.(func(fyne.FileReadCloser, error)), fd.parent) {
+			return
+		}
+		fd.dialog = showFileDialog(false, fd.callback, fd.parent, fd.filter)
+		return
+	}
+	if fileSaveOSOverride(fd.callback.(func(fyne.FileWriteCloser, error)), fd.parent) {
 		return
 	}
-	showFileDialog(false, callback, parent)
+	fd.dialog = showFileDialog(true, fd.callback, fd.parent, fd.filter)
+}
+
+// NewExtensionFileFilter takes a string slice of extensions with a leading . and creates a filter for the file dialog.
+// Example: .jpg, .mp3, .txt, .sh
+func NewExtensionFileFilter(extensions []string) FileFilter {
+	return &extensionFileFilter{extensions: extensions}
+}
+
+// NewMimeTypeFileFilter takes a string slice of mimetypes, including globs, and creates a filter for the file dialog.
+// Example: image/*, audio/mp3, text/plain, application/*
+func NewMimeTypeFileFilter(mimeTypes []string) FileFilter {
+	return &mimeTypeFileFilter{mimeTypes: mimeTypes}
 }
 
-// ShowFileSave shows a file dialog allowing the user to choose a file to save to (new or overwrite).
+// NewFileOpenDialog creates a file dialog allowing the user to choose a file to open.
+// The dialog will appear over the window specified.
+func NewFileOpenDialog(callback func(fyne.FileReadCloser, error), parent fyne.Window) *FileDialog {
```

Comment:
dialog.NewFileOpenDialog is repetition and the others haven't got ...Dialog suffix, is it needed?